### PR TITLE
Remove wrapper RequestMetadata type.

### LINKED ISSUE #94: Remove RequestMetadata type
This struct stores the scalar across Request/Finalize calls, which both happen on the client. There's no need to serialize or send this value on the wire anywhere, so we should just defer its representation to the underlying implementation. Given that the OPRF API is already a dependency we expose, I suggest we replace RequestMetadata with the actual type `Scalar`. 

While we're in here, we might also consider removing superfluous wrappers around OPRF types (like `RegistrationRequest` and `CredentialRequest`). 

@kevinlewi, what do you think?

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #94.

cc @kevinlewi

### CODE REVIEW

File: draft-irtf-cfrg-opaque.md [hunk 1]
```diff
@@ -796,18 +780,18 @@ Steps:
...
+RecoverCredentials(pwdU, blind, request, response)
 
 Parameters:
 - params, the MHF parameters established out of band
 - Nh, the output size of the Hash function
 
 Input:
 - pwdU, an opaque byte string containing the user's password
-- metadata, a RequestMetadata structure
+- blind, a RequestMetadata structure
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:794]: "a RequestMetadata structure" should be replaced with "an OPRF Scalar value"

File: draft-irtf-cfrg-opaque.md [hunk 2]
```diff
@@ -517,33 +517,24 @@ See {{validation}} for more details.
...
-} RequestMetadata;
-~~~
-
-data_blind
-: An encoded OPRF scalar element.
-
-~~~
-struct {
-    opaque data<0..2^16-1>;
+    SerializedElement data;
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:529]: Where is SerializedElement defined? Perhaps a reference to an RFC #?

File: draft-irtf-cfrg-opaque.md (hunk 1 above)
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:794]: Missed this one -- thanks!

File: draft-irtf-cfrg-opaque.md (hunk 2 above)
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:530]: I clarified that this is an alias for a serialized group element, i.e., output of `Serialize()`.

**Review by kevinlewi — APPROVED** [review]
Looks good!

**Pull request merged**